processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -188,6 +188,22 @@ public BeanMappingMethod build() {
                             Message.BEANMAPPING_NOT_ASSIGNABLE, resultType, method.getResultType()
                         );
                     }
+                    else if ( resultType.isAbstract() ) {
+                        ctx.getMessager().printMessage(
+                            method.getExecutable(),
+                            beanMappingPrism.mirror,
+                            Message.BEANMAPPING_ABSTRACT,
+                            resultType,
+                            method.getResultType()
+                        );
+                    }
+                }
+                else if ( !method.isUpdateMethod() && method.getReturnType().isAbstract() ) {
```

Comment:
There is one general place where interfaces are handled. That is in [here](https://github.com/mapstruct/mapstruct/blob/6b07dda829788387abef3ff8574066553ad7b646/processor/src/main/java/org/mapstruct/ap/internal/processor/MapperCreationProcessor.java#L538) in the `MapperCreationProcessor`. That one handles the other types as well.